certbot-apache/certbot_apache/configurator.py

```diff
@@ -959,9 +968,100 @@ def _copy_create_ssl_vhost_skeleton(self, avail_fp, ssl_fp):
+                    vhost.filep, ssl_fp), reporter.MEDIUM_PRIORITY)
+        self.aug.set("/augeas/files%s/mtime" % (self._escape(ssl_fp)), "0")
+        self.aug.set("/augeas/files%s/mtime" % (self._escape(vhost.filep)), "0")
+
+    def _prepare_ssl_skeleton_contents(self, contents):
+        """ Helper function for _copy_create_ssl_vhost_skeleton to prepare the
+        new HTTPS VirtualHost contents. Currently disabling the rewrites """
+
+        result = []
+        sift = False
+        contents = iter(contents)
+
+        comment = ("# Some rewrite rules in this file were "
+                   "disabled on your HTTPS site,\n"
+                   "# because they have the potential to create "
+                   "redirection loops.\n")
+
+        for line in contents:
+            A = line.lower().lstrip().startswith("rewritecond")
+            B = line.lower().lstrip().startswith("rewriterule")
+
+            if not (A or B):
+                result.append(line)
+                continue
+
+            # A RewriteRule that doesn't need filtering
+            if B and not self._sift_rewrite_rule(line):
+                result.append(line)
+                continue
+
+            # A RewriteRule that does need filtering
+            if B and self._sift_rewrite_rule(line):
+                if not sift:
+                    result.append(comment)
+                    sift = True
+                result.append("# " + line)
+                continue
+
+            # We save RewriteCond(s) and their corresponding
+            # RewriteRule in 'chunk'.
+            # We then decide whether we comment out the entire
+            # chunk based on its RewriteRule.
+            chunk = []
+            if A:
+                chunk.append(line)
+                line = next(contents)
+
+                # RewriteCond(s) must be followed by one RewriteRule
+                while not line.lower().lstrip().startswith("rewriterule"):
+                    chunk.append(line)
+                    line = next(contents)
+
+                # Now, current line must start with a RewriteRule
+                chunk.append(line)
+
+                if self._sift_rewrite_rule(line):
+                    if not sift:
+                        result.append(comment)
+                        sift = True
+
+                    result.append('\n'.join(
+                        ['# ' + l for l in chunk]))
+                    continue
+                else:
+                    result.append('\n'.join(chunk))
+                    continue
+        return result, sift
+
+    def _get_vhost_block(self, vhost):
+        """ Helper method to get VirtualHost contents from the original file.
+        This is done with help of augeas span, which returns the span start and
+        end positions
+
+        :returns: `list` of VirtualHost block content lines without closing tag
+        """
+
+        try:
+            span_val = self.aug.span(vhost.path)
```

Comment:
The configuration directives inside blocks apply globally, so it's not that big of an issue. However people be overwriting some values in nested blocks containing VirtualHosts. Then again people could similarly have things like:
```
Listen 80
<VirtualHost *:80>
...
</VirtualHost>
```
as an example, which would result to apache failing to address already in use error. So it kind of goes both ways.

I think we should just copy the VirtualHost itself. Linux distributions provide examples for, and encourage people to use a structure where VirtualHosts are sitting on root level of the configuration file. The IfModules and IfDefines are pretty much just in place to prevent loading of certain (default) config blocks if modules are not enabled. 

Also I doubt people play around with the modules a lot. Basically, if you have your configuration running, you do have the modules in anyway, and the point where it would break (in configtest syntax check fail way of breaking) would be when you would disable a module manually later on.

If anything, I would say we'd need to put more effort into active module parsing and not peeking in to configuration structures under disabled IfModule statements, hence not offering the disabled domains as installation candidates by Certbot.

In a world where we wouldn't need to revisit the configuration ever again, or give user a clear indication of the new vhost file we created - we'd create the new VirtualHost in the original file, right beside the original one, hence leveraging all the nested config directives in place.